mlflow/store/model_registry/abstract_store.py

```diff
@@ -462,96 +461,200 @@ def _await_model_version_creation_impl(self, mv, await_creation_for, hint=""):
         catalog_name: Optional[str] = None,
         schema_name: Optional[str] = None,
-    ) -> PagedList[PromptInfo]:
+    ):
         """
-        Search for prompts that satisfy the filter criteria.
+        Search for prompts in the registry.
+
+        Default implementation for non-Unity Catalog registries: searches RegisteredModels
+        with prompt tags. Unity Catalog stores override this method.
 
         Args:
-            filter_string: Filter query string.
+            filter_string: Filter query string, defaults to searching for all prompts.
             max_results: Maximum number of prompts desired.
-            order_by: List of column names with ASC|DESC annotation.
-            page_token: Token specifying the next page of results.
-            catalog_name: Unity Catalog catalog name (for UC registries).
-            schema_name: Unity Catalog schema name (for UC registries).
+            order_by: List of order-by clauses.
+            page_token: Pagination token for requesting subsequent pages.
+            catalog_name: Catalog name (Unity Catalog only).
+            schema_name: Schema name (Unity Catalog only).
 
         Returns:
-            A PagedList of prompt objects that satisfy the search expressions.
+            A PagedList of PromptInfo objects.
         """
+        from mlflow.prompt.constants import IS_PROMPT_TAG_KEY
+
+        # Build filter to only include prompts (use backticks for tag key with dots)
+        prompt_filter = f"tags.`{IS_PROMPT_TAG_KEY}` = 'true'"
+        if filter_string:
+            prompt_filter = f"{prompt_filter} AND {filter_string}"
+
+        # Search registered models with prompt filter
+        registered_models = self.search_registered_models(
+            filter_string=prompt_filter,
+            max_results=max_results,
+            order_by=order_by,
+            page_token=page_token,
+        )
+
+        # For backward compatibility, return RegisteredModel objects with latest_versions
+        # The traditional search_prompts() expects objects with latest_versions attribute
+        for rm in registered_models:
+            # Ensure the registered model has latest_versions populated
+            if not hasattr(rm, "latest_versions") or rm.latest_versions is None:
+                # Get the latest versions for this prompt
+                from mlflow.entities.model_registry.model_version_stages import ALL_STAGES
+
+                try:
+                    latest_versions = self.get_latest_versions(rm.name, stages=ALL_STAGES)
+                    rm.latest_versions = latest_versions
+                except Exception:
+                    rm.latest_versions = []
+
+        return registered_models
 
-    @abstractmethod
     def delete_prompt(self, name: str) -> None:
         """
-        Delete a prompt.
+        Delete a prompt from the registry.
+
+        Default implementation for non-Unity Catalog registries: deletes the underlying
```

Comment:
Can we avoid mentioning unity catalog in the abstract store methods? We can just say that "other stores" may override this method.